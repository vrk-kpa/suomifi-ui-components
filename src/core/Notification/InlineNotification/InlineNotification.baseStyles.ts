import { css } from 'styled-components';
import { SuomifiTheme } from '../../theme';
import { font } from '../../theme/reset';
import { baseNotificationBaseStyles } from '../BaseNotification/BaseNotification.baseStyles';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${baseNotificationBaseStyles(theme)}
  & .fi-notification_label {
    ${font(theme)('bodySemiBold')}
    margin-bottom: ${theme.spacing.xxs};
  }

  /** Inline variant styles  */
  &.fi-notification--inline {
    padding: 5px 0px 4px 0px;
    .fi-notification_icon-wrapper {
      flex: 0;
    }
  }
`;
