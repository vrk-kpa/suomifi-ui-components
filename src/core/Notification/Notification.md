The `<Notification>` component provides users noteworthy information about a specific part of a website or an action they made.

Notification should not be used dynamically. For dynamic use cases, use the [InlineAlert](./#/Components/InlineAlert) component instead.

If you need to show site-wide information at the top of the page, use the [Alert](./#/Components/Alert) component instead.

Examples:

- [Basic use](./#/Components/Notification?id=basic-use)
- [Heading levels](./#/Components/Notification?id=heading-levels)
- [Action elements](./#/Components/Notification?id=action-elements)
- [Error status](./#/Components/Notification?id=error-status)
- [Without close button](./#/Components/Notification?id=without-close-button)
- [Small screen](./#/Components/Notification?id=small-screen)

<div style="margin-bottom: 40px">
  [Props & methods](./#/Components/Notification?id=props--methods)
</div>

### Basic use

Provide the `closeText` prop to label the close button.

If `headingText` is not provided, text content is used to label the component's `role="region"`. You can also use the `regionAriaLabel` prop to label the region explicitly.

```js
import { Notification } from 'suomifi-ui-components';
import { useState } from 'react';

const [showNotification, setShowNotification] = useState(true);

showNotification && (
  <Notification
    closeText="Close"
    headingText="Translation arrived (Swedish)"
    onCloseButtonClick={() => setShowNotification(false)}
  >
    Inspect the translation and edit it if necessary
  </Notification>
);
```

### Heading levels

Use the `headingVariant` prop to change the semantics of the heading. Visual styling of the heading is fixed.

```js
import { Notification } from 'suomifi-ui-components';
import { useState } from 'react';

const [showNotification, setShowNotification] = useState(true);

showNotification && (
  <Notification
    closeText="Close"
    headingText="Translation arrived (Swedish)"
    headingVariant="h4"
  >
    Inspect the translation and edit it if necessary
  </Notification>
);
```

### Action elements

Use the `actionElements` prop to render interactive elements below the component's main text content.

```js
import { Notification, Button } from 'suomifi-ui-components';
import { useState } from 'react';

const [showNotification, setShowNotification] = useState(true);

showNotification && (
  <Notification
    closeText="Close"
    headingText="Translation arrived (Swedish)"
    actionElements={
      <Button variant="secondary">Inspect translation</Button>
    }
  >
    Inspect the translation and edit it if necessary
  </Notification>
);
```

### Error status

Use `status="error"` to apply error styles to the Notification. Use a descripitve heading to emphasize that this is an error notification.

```js
import { Notification } from 'suomifi-ui-components';
import { useState } from 'react';

const [showNotification, setShowNotification] = useState(true);

showNotification && (
  <Notification
    closeText="Close"
    headingText="Signing is not possible"
    status="error"
  >
    You cannot sign the document with an expired identification
    document
  </Notification>
);
```

### Without close button

In some cases, like when the notification is the sole content on the page, the notification may be rendered as a non-interactive version by disabling the close button. This can be done via the `showCloseButton` prop.

```js
import { Notification } from 'suomifi-ui-components';

<Notification
  showCloseButton={false}
  headingText="Content not found"
  status="error"
>
  The resource you were looking for could not be found.
</Notification>;
```

### Small screen

Use the `smallScreen` prop to toggle small screen styles.

```js
import { Notification } from 'suomifi-ui-components';
import { useState } from 'react';

const [showNotification, setShowNotification] = useState(true);

showNotification && (
  <Notification
    closeText="Close"
    headingText="Translation arrived (Swedish)"
    smallScreen
    onCloseButtonClick={() => setShowNotification(false)}
  >
    Inspect the translation and edit it if necessary
  </Notification>
);
```

### Props & methods

Notification component supports [margin props](./#/Spacing/Margin%20props) for spacing.
