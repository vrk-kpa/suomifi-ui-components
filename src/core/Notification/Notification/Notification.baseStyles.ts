import { css } from 'styled-components';
import { SuomifiTheme } from '../../theme';
import { font } from '../../theme/reset';
import { baseNotificationBaseStyles } from '../BaseNotification/BaseNotification.baseStyles';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${baseNotificationBaseStyles(theme)}
  & .fi-notification_close-button-wrapper {
    flex: 1 0 auto;
    display: flex;
    flex-wrap: nowrap;
    box-sizing: border-box;
    margin-top: 7px;
    margin-right: ${theme.spacing.xs};
    margin-bottom: ${theme.spacing.insetM};
  }

  & .fi-notification_close-button {
    ${font(theme)('bodyTextSmall')}
    height: 40px;
    min-width: 40px;
    padding: 7px 8px;
    border: 1px solid transparent;
    border-radius: ${theme.radius.basic};

    &:focus-visible {
      outline: 0;
      position: relative;

      &:after {
        ${theme.focus.absoluteFocus}
      }
    }
    &:active {
      background: ${theme.gradients.whiteBaseToDepthLight1};
    }
    &:hover {
      border-color: ${theme.colors.blackBase};
    }

    & .fi-icon {
      width: 14px;
      height: 14px;
      margin-left: ${theme.spacing.xxs};
      margin-top: ${theme.spacing.xxs};
      transform: translateY(1px);
    }
  }

  /** Small screen variant styles */
  &--small-screen {
    & .fi-notification-close-button-wrapper {
      justify-content: flex-end;
      margin: 0;
      & .fi-icon {
        margin-right: ${theme.spacing.xxs};
      }
    }
  }
`;
