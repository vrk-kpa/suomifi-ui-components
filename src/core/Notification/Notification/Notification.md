```js
import {
  Notification,
  InlineNotification,
  Button
} from 'suomifi-ui-components';

<>
  <Notification closeText="Close" smallScreen>
    The service will be temporarily unavailable on 5.6.2021 at 21.00 –
    23.59 due to maintenance.
  </Notification>

  <Notification
    closeText="Close"
    labelText="Maintenance notification"
  >
    This is a long notification text. Lorem ipsum dolor sit amet,
    consectetur adipiscing elit. Morbi est nulla, rhoncus ac odio at,
    blandit ornare mi. Suspendisse potenti. Vestibulum ante ipsum
    primis in faucibus orci luctus et ultrices posuere cubilia curae;
    Suspendisse efficitur et turpis eget tempor. Nam sed lacus ut
    lorem feugiat tincidunt. Mauris quis nisi placerat, convallis
    turpis nec, auctor massa. Nam ornare tortor id tortor tristique,
    in semper augue fermentum. Nulla at pharetra leo, a ullamcorper
    metus. Donec molestie velit tristique feugiat interdum. Donec
    ultrices efficitur pellentesque. Sed luctus ac metus sed rhoncus.
  </Notification>

  <InlineNotification status="error">
    <p>Something went wrong. Please try again in a moment.</p>
    <Button fullWidth variant="secondary">
      Button fullWidth variant="secondary"
    </Button>
  </InlineNotification>

  <Notification status="error" closeText="Long close text">
    Something went wrong. Please try again in a moment.
  </Notification>

  <Notification status="success" closeText="Close">
    This is a long notification text. Lorem ipsum dolor sit amet,
    consectetur adipiscing elit. Morbi est nulla, rhoncus ac odio at,
    blandit ornare mi. Suspendisse potenti. Vestibulum ante ipsum
    primis in faucibus orci luctus et ultrices posuere cubilia curae;
    Suspendisse efficitur et turpis eget tempor. Nam sed lacus ut
    lorem feugiat tincidunt.
  </Notification>

  <InlineNotification status="success" labelText="Warning">
    Password Uptated.
  </InlineNotification>

  <InlineNotification labelText="Long label Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi est nulla, rhoncus ac odio at">
    The service will be temporarily unavailable on 5.6.2021 at 21.00 –
    23.59 due to maintenance.
  </InlineNotification>
</>;
```
