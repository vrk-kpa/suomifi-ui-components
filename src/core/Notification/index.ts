export { Notification, NotificationProps } from './Notification/Notification';
export {
  InlineNotification,
  InlineNotificationProps,
} from './InlineNotification/InlineNotification';
