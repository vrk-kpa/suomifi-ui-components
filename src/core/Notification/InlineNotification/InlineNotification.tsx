import React, { Component, forwardRef } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import { HtmlDiv, HtmlDivWithRef } from '../../../reset';
import { Icon } from '../../Icon/Icon';
import {
  BaseNotificationProps,
  notificationClassNames,
  baseClassName,
} from '../BaseNotification/BaseNotification';
import { AutoId } from '../../utils/AutoId/AutoId';
import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
import { baseStyles } from './InlineNotification.baseStyles';

export interface InlineNotificationProps extends BaseNotificationProps {
  /** Label for the Notification */
  labelText?: string;
}

interface InnerRef {
  forwardedRef: React.RefObject<HTMLDivElement>;
}

class BaseInlineNotification extends Component<
  InlineNotificationProps & InnerRef
> {
  render() {
    const {
      className,
      status = 'neutral',
      ariaLiveMode = 'assertive',
      labelText,
      children,
      id,
      ...passProps
    } = this.props;

    const variantIcon = status === 'neutral' ? 'info' : status;
    return (
      <HtmlDivWithRef
        as="section"
        {...passProps}
        className={classnames(
          baseClassName,
          notificationClassNames.inline,
          className,
          {
            [`${baseClassName}--${status}`]: !!status,
          },
        )}
      >
        <HtmlDiv className={notificationClassNames.styleWrapper}>
          <HtmlDiv className={notificationClassNames.iconWrapper}>
            <Icon icon={variantIcon} className={notificationClassNames.icon} />
          </HtmlDiv>

          <HtmlDiv
            className={notificationClassNames.textContentWrapper}
            id={id}
            aria-live={ariaLiveMode}
          >
            {labelText && (
              <HtmlDiv className={notificationClassNames.label}>
                {labelText}
              </HtmlDiv>
            )}
            <HtmlDiv className={notificationClassNames.content}>
              {children}
            </HtmlDiv>
          </HtmlDiv>
        </HtmlDiv>
      </HtmlDivWithRef>
    );
  }
}

const StyledInlineNotification = styled(
  (props: InlineNotificationProps & InnerRef & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseInlineNotification {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)};
`;

export const InlineNotification = forwardRef(
  (props: InlineNotificationProps, ref: React.RefObject<HTMLDivElement>) => {
    const { id: propId, ...passProps } = props;
    return (
      <SuomifiThemeConsumer>
        {({ suomifiTheme }) => (
          <AutoId id={propId}>
            {(id) => (
              <StyledInlineNotification
                forwardedRef={ref}
                theme={suomifiTheme}
                id={id}
                {...passProps}
              />
            )}
          </AutoId>
        )}
      </SuomifiThemeConsumer>
    );
  },
);
