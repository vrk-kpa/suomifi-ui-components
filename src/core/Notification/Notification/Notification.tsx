import React, { Component, forwardRef } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import {
  HtmlDiv,
  HtmlDivWithRef,
  HtmlButton,
  HtmlButtonProps,
} from '../../../reset';
import { Icon } from '../../Icon/Icon';
import {
  BaseNotificationProps,
  notificationClassNames,
  baseClassName,
} from '../BaseNotification/BaseNotification';
import { AutoId } from '../../utils/AutoId/AutoId';
import { getConditionalAriaProp } from '../../../utils/aria';
import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
import { baseStyles } from './Notification.baseStyles';

export interface NotificationProps extends BaseNotificationProps {
  /** Text to label the close button. Visible + `aria-label` in regular size and only used as `aria-label` in small screen variant */
  closeText: string;
  /** Click handler for the close button */
  onCloseButtonClick?: () => void;
  /** Custom props passed to the close button */
  closeButtonProps?: Omit<HtmlButtonProps, 'onClick' | 'aria-label'>;
  /** Use small screen styling */
  smallScreen?: boolean;
}

interface InnerRef {
  forwardedRef: React.RefObject<HTMLDivElement>;
}

class BaseNotification extends Component<NotificationProps & InnerRef> {
  render() {
    const {
      className,
      status = 'neutral',
      ariaLiveMode = 'assertive',
      labelText,
      children,
      id,
      closeText,
      onCloseButtonClick,
      smallScreen,
      closeButtonProps = {},
      ...passProps
    } = this.props;

    const {
      className: customCloseButtonClassName,
      'aria-describedby': closeButtonPropsAriaDescribedBy,
      ...closeButtonPassProps
    } = closeButtonProps;

    const variantIcon = status === 'neutral' ? 'info' : status;
    return (
      <HtmlDivWithRef
        as="section"
        {...passProps}
        className={classnames(baseClassName, className, {
          [`${baseClassName}--${status}`]: !!status,
          [notificationClassNames.smallScreen]: !!smallScreen,
        })}
      >
        <HtmlDiv className={notificationClassNames.styleWrapper}>
          <HtmlDiv className={notificationClassNames.iconWrapper}>
            <Icon icon={variantIcon} className={notificationClassNames.icon} />
          </HtmlDiv>

          <HtmlDiv
            className={notificationClassNames.textContentWrapper}
            id={id}
            aria-live={ariaLiveMode}
          >
            <HtmlDiv className={notificationClassNames.content}>
              {children}
            </HtmlDiv>
          </HtmlDiv>
          <HtmlDiv className={notificationClassNames.closeButtonWrapper}>
            <HtmlButton
              className={classnames(
                notificationClassNames.closeButton,
                customCloseButtonClassName,
              )}
              {...getConditionalAriaProp('aria-describedby', [
                id,
                closeButtonPropsAriaDescribedBy,
              ])}
              onClick={onCloseButtonClick}
              {...getConditionalAriaProp('aria-label', [closeText])}
              {...closeButtonPassProps}
            >
              {!smallScreen ? closeText?.toUpperCase() : ''}
              <Icon icon="close" />
            </HtmlButton>
          </HtmlDiv>
        </HtmlDiv>
      </HtmlDivWithRef>
    );
  }
}

const StyledNotification = styled(
  (props: NotificationProps & InnerRef & SuomifiThemeProp) => {
    const { theme, ...passProps } = props;
    return <BaseNotification {...passProps} />;
  },
)`
  ${({ theme }) => baseStyles(theme)};
`;

export const Notification = forwardRef(
  (props: NotificationProps, ref: React.RefObject<HTMLDivElement>) => {
    const { id: propId, ...passProps } = props;
    return (
      <SuomifiThemeConsumer>
        {({ suomifiTheme }) => (
          <AutoId id={propId}>
            {(id) => (
              <StyledNotification
                forwardedRef={ref}
                theme={suomifiTheme}
                id={id}
                {...passProps}
              />
            )}
          </AutoId>
        )}
      </SuomifiThemeConsumer>
    );
  },
);
