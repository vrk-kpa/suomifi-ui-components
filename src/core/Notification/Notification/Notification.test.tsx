import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Notification } from './Notification';
import { InlineNotification } from '../InlineNotification/InlineNotification';
import { axeTest } from '../../../utils/test';

describe('children', () => {
  const notificationWithElementChild = (
    <Notification closeText="Close">
      <div data-testid="test-div">Test notification</div>
    </Notification>
  );

  it('has the given content', () => {
    const { container } = render(notificationWithElementChild);
    expect(container.firstChild).toHaveTextContent('Test notification');
  });

  it('should match snapshot', () => {
    const { container } = render(notificationWithElementChild);
    expect(container).toMatchSnapshot();
  });
});

describe('classnames', () => {
  const customClassNotification = (
    <Notification closeText="Close" className="custom-class">
      Test content
    </Notification>
  );

  it('contains base classname', () => {
    const { container } = render(customClassNotification);
    expect(container.firstChild).toHaveClass('fi-notification');
  });

  it('contains custom classname', () => {
    const { container } = render(customClassNotification);
    expect(container.firstChild).toHaveClass('custom-class');
  });
});

describe('props', () => {
  const NotificationWithButtonProps = (
    <Notification
      closeText="Close"
      closeButtonProps={{ 'aria-labelledby': 'test-element', disabled: true }}
    >
      Testcontent
    </Notification>
  );

  it('Button should have given props', () => {
    const { getByRole } = render(NotificationWithButtonProps);
    expect(getByRole('button')).toHaveAttribute(
      'aria-labelledby',
      'test-element',
    );
    expect(getByRole('button')).toHaveAttribute('disabled');
    expect(getByRole('button')).toHaveTextContent('CLOSE');
  });

  const InlineNotificationComponent = (
    <InlineNotification labelText="Inline Notification label">
      Testcontent
    </InlineNotification>
  );

  it('Inline component should contain given labelText', () => {
    const { getByText } = render(InlineNotificationComponent);
    const label = getByText('Inline Notification label');
    expect(label).toHaveClass('fi-notification_label');
  });

  const ErrorNotification = (
    <Notification
      status="error"
      className="notification-test-class"
      closeText="Close"
      smallScreen
    >
      Testcontent
    </Notification>
  );

  it('should have status and smallScreen prop specific classNames', () => {
    const { container } = render(ErrorNotification);
    expect(container.firstChild).toHaveClass('fi-notification--error');
    expect(container.firstChild).toHaveClass('fi-notification--small-screen');
  });
  it('should have given className as props', () => {
    const { container } = render(ErrorNotification);
    expect(container.firstChild).toHaveClass('notification-test-class');
  });

  const NotificationWithDefaultAriaLiveMode = (
    <Notification id="testId" closeText="Close" ariaLiveMode="off">
      Testcontent
    </Notification>
  );

  it('should have specified aria-live mode', () => {
    const { container } = render(NotificationWithDefaultAriaLiveMode);
    expect(container.querySelector('#testId')).toHaveClass(
      'fi-notification_text-content-wrapper',
    );
    expect(container.querySelector('#testId')).toHaveAttribute(
      'aria-live',
      'off',
    );
  });
  test('onClick event is called when clicked', () => {
    const mockClick = jest.fn();
    const { getByRole } = render(
      <Notification closeText="Close" onClick={mockClick}>
        Test content
      </Notification>,
    );
    const button = getByRole('button');
    fireEvent.click(button);
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
});

describe('accessibility', () => {
  const TestNotification = (
    <Notification closeText="Close">Testcontent</Notification>
  );
  test('should not have basic accessibility issues', axeTest(TestNotification));
});
