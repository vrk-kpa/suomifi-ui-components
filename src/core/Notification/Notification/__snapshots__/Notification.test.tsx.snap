// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`children should match snapshot 1`] = `
.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c4:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c4::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c4::-webkit-inner-spin-button {
  height: auto;
}

.c4::-webkit-outer-spin-button {
  height: auto;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c3 {
  display: inline-block;
  vertical-align: baseline;
}

.c3 .fi-icon-base-fill {
  fill: currentColor;
}

.c3 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1.fi-notification {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  width: 100%;
  box-shadow: 0px 4px 8px 0px rgba(41,41,41,0.2);
  border-radius: 4px;
}

.c1.fi-notification .fi-notification_style-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1.fi-notification .fi-notification_icon-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-left: 15px;
  margin-top: 16px;
}

.c1.fi-notification .fi-notification_icon-wrapper .fi-notification_icon {
  height: 24px;
  width: 24px;
}

.c1.fi-notification .fi-notification_text-content-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 3 1 auto;
  -ms-flex: 3 1 auto;
  flex: 3 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 20px;
  margin-top: 20px;
  margin-bottom: 15px;
}

.c1.fi-notification .fi-notification_text-content-wrapper .fi-notification_content {
  vertical-align: middle;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-notification--neutral {
  border-top: 5px solid hsl(196,77%,44%);
}

.c1.fi-notification--neutral .fi-notification_icon-wrapper .fi-icon .fi-icon-base-fill {
  fill: hsl(212,63%,45%);
}

.c1.fi-notification--error {
  border-top: 5px solid hsl(3,59%,48%);
}

.c1.fi-notification--error .fi-notification_icon-wrapper .fi-icon .fi-icon-base-fill {
  fill: hsl(3,59%,48%);
}

.c1.fi-notification--success {
  border-top: 5px solid hsl(166,90%,34%);
}

.c1.fi-notification--success .fi-notification_icon-wrapper .fi-icon .fi-icon-base-fill {
  fill: hsl(166,90%,34%);
}

.c1 .fi-notification_close-button-wrapper {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
  margin-top: 7px;
  margin-right: 10px;
  margin-bottom: 8px;
}

.c1 .fi-notification_close-button {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  height: 40px;
  min-width: 40px;
  padding: 7px 8px;
  border: 1px solid transparent;
  border-radius: 2px;
}

.c1 .fi-notification_close-button:focus-visible {
  outline: 0;
  position: relative;
}

.c1 .fi-notification_close-button:focus-visible:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1 .fi-notification_close-button:active {
  background: linear-gradient(-180deg,hsl(202,7%,80%) 0%,rgba(255,255,255,0) 100%);
}

.c1 .fi-notification_close-button:hover {
  border-color: hsl(0,0%,16%);
}

.c1 .fi-notification_close-button .fi-icon {
  width: 14px;
  height: 14px;
  margin-left: 5px;
  margin-top: 5px;
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c1--small-screen .fi-notification-close-button-wrapper {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 0;
}

.c1--small-screen .fi-notification-close-button-wrapper .fi-icon {
  margin-right: 5px;
}

<div>
  <section
    class="c0 fi-notification c1 fi-notification--neutral"
  >
    <div
      class="c2 fi-notification_style-wrapper"
    >
      <div
        class="c2 fi-notification_icon-wrapper"
      >
        <svg
          aria-hidden="true"
          class="fi-icon c3 fi-notification_icon"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="fi-icon-base-fill"
            d="M12 0c6.617 0 12 5.383 12 12s-5.383 12-12 12S0 18.617 0 12 5.383 0 12 0zm0 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm.003 8a1 1 0 011 1v6h1a1 1 0 010 2H10a1 1 0 010-2h1v-5h-1a1 1 0 010-2h2.003zm-.501-5a1.5 1.5 0 01.145 2.994L11.502 8H11.5a1.5 1.5 0 110-3h.002z"
            fill="#292929"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        aria-live="assertive"
        class="c2 fi-notification_text-content-wrapper"
        id="3"
      >
        <div
          class="c2 fi-notification_content"
        >
          <div>
            Test notification
          </div>
        </div>
      </div>
      <div
        class="c2 fi-notification_close-button-wrapper"
      >
        <button
          aria-describedby="3"
          aria-label="Close"
          class="c4 fi-notification_close-button"
          type="button"
        >
          CLOSE
          <svg
            aria-hidden="true"
            class="fi-icon c3"
            focusable="false"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="fi-icon-base-fill"
              d="M4.75 3.3L12 10.55l7.25-7.25a1.026 1.026 0 011.45 1.45L13.45 12l7.25 7.25a1.026 1.026 0 01-1.45 1.45L12 13.45 4.75 20.7a1.026 1.026 0 01-1.45-1.45L10.55 12 3.3 4.75A1.026 1.026 0 014.75 3.3z"
              fill="#292929"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </section>
</div>
`;
